ports/servoshell/desktop/cli.rs

```diff
@@ -8,13 +8,32 @@ use getopts::Options;
 use log::error;
 use servo::config::opts::{self, ArgumentParsingResult};
 use servo::servo_config::pref;
+use tracing_perfetto::PerfettoLayer;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
+use tracing_subscriber::EnvFilter;
 
 use crate::desktop::app::App;
 use crate::panic_hook;
 
 pub fn main() {
     crate::crash_handler::install();
 
+    let file = std::fs::File::create("servo.pftrace").unwrap();
+    let perfetto_layer = PerfettoLayer::new(std::sync::Mutex::new(file));
+    // Set up a custom tracing subscriber that behaves like the default fmt
+    // subscriber, but also lets us insert our own layers for things like
+    // compositor-to-constellation and script-to-constellation logging.
+    // <https://tokio.rs/tokio/topics/tracing>
+    // <https://docs.rs/tracing-subscriber/0.3.18/tracing_subscriber/fmt/index.html#composing-layers>
+    let env_filter_layer = EnvFilter::from_default_env();
+    let fmt_layer = tracing_subscriber::fmt::layer();
+    tracing_subscriber::registry()
+        .with(env_filter_layer)
+        .with(fmt_layer)
+        .with(perfetto_layer)
+        .init();
```

Comment:
Should this be turned on by a command-line argument which accepts an output filename?

Reply:
Oh, I have been thinking about this, but my concern comes from a different reason. This is really slowing down the build. On Mac, it takes around 5 minutes to build.

I've been discussing adding a flag with @delan and will wait for her feedback as well.

Reply:
yeah, there’s a smaller but still significant regression in build time on linux too (before [135s](https://github.com/user-attachments/files/16761228/cargo-timing-20240827T110216Z.html.gz), after [160s](https://github.com/user-attachments/files/16761233/cargo-timing-20240827T105843Z.html.gz)). what if we gate this behind a `tracing` or `perfetto` feature for now?

Reply:
sounds good to me. I have added the `tracing` feature flag in [519e954](https://github.com/servo/servo/pull/33188/commits/519e954d58413b8f9513d30a79ae694a48bc0c1f). I guess moving forward, we can modify it if needed.